homeassistant/components/nina/config_flow.py

```diff
@@ -0,0 +1,171 @@
+"""Config flow for Nina integration."""  # pylint: disable=R0801
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol  # pylint: disable=E0401
+from pynina import ApiError, Nina  # pylint: disable=E0401
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import (  # pylint: disable=W0611
+    _LOGGER,
+    CONF_FILTER_CORONA,
+    CONF_MESSAGE_SLOTS,
+    CONF_REGIONS,
+    DOMAIN,
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for NINA."""
+
+    VERSION: int = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        super().__init__()
+        self._all_region_codes_sorted: dict[str, str] = {}  # pylint: disable=E1136
+        self.regions_a: dict[str, Any] = {}  # pylint: disable=E1136
+        self.regions_b: dict[str, Any] = {}  # pylint: disable=E1136
+        self.regions_c: dict[str, Any] = {}  # pylint: disable=E1136
+        self.regions_d: dict[str, Any] = {}  # pylint: disable=E1136
+        self.regions_e: dict[str, Any] = {}  # pylint: disable=E1136
+        self.regions_f: dict[str, Any] = {}  # pylint: disable=E1136
+
+    async def async_step_user(  # pylint: disable=R0914,R0912
+        self: ConfigFlow,  # pylint: disable=C0330
+        user_input: dict[str, Any] | None = None,  # pylint: disable=E1136,C0326,C0330
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, Any] = {}  # pylint: disable=E1136
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        has_error: bool = False
+
+        if len(self._all_region_codes_sorted) == 0:  # pylint: disable=R1702
+            try:
+                nina: Nina = Nina()
+
+                self._all_region_codes_sorted = self.swap_key_value(
+                    await nina.getAllRegionalCodes()
+                )
+
+            except ApiError as err:
+                _LOGGER.warning("NINA setup error: %s", err)
+                errors["base"] = "cannot_connect"
+                has_error = True
+            except Exception as err:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception: %s", err)
+                errors["base"] = "unknown"
+                has_error = True
```

Comment:
you should abort here, right?

If the API does not return anything it does not make sense to show the form at all.

Reply:
Remove `has_error` completely